antarest/study/business/binding_constraint_management.py

```diff
@@ -178,167 +391,329 @@ def parse_constraint(key: str, value: str, char: str, new_config: BindingConstra
         return False
 
     @staticmethod
-    def process_constraint(constraint_value: Dict[str, Any], version: int) -> BindingConstraintConfigType:
-        args = {
-            "id": constraint_value["id"],
-            "name": constraint_value["name"],
-            "enabled": constraint_value["enabled"],
-            "time_step": constraint_value["type"],
-            "operator": constraint_value["operator"],
-            "comments": constraint_value.get("comments", None),
-            "filter_year_by_year": constraint_value.get("filter-year-by-year", ""),
-            "filter_synthesis": constraint_value.get("filter-synthesis", ""),
-            "constraints": None,
+    def constraint_model_adapter(constraint: ConstraintInput, version: int) -> ConstraintOutput:
```

Comment:
Wrong typing for `constraint`, should be:

```python
constraint: t.Mapping[str, t.Any]]
```

Reply:
Yes their some typing issues, I am fixing this as well as tests